scripts/talos/bumpSimorghPackages/updatePackageJson/index.js

```diff
@@ -0,0 +1,30 @@
+const commitRemoteChanges = require('../../../utilities/commitRemoteChanges');
+
+const mergeUpdates = (packageJson, publishedPackages) => {
+  const newPackageJson = { ...packageJson };
+  Object.keys(publishedPackages).forEach(key => {
```

Comment:
I think the `mergeUpdates` abstraction should be the callback function in the forEach here. It's also possible to build up a single object using a reduce function instead of creating an object to mutate. An example of this could be:

```js
const mergeUpdates = publishedPackages => (packageJson, key) => {
  const { dependencies, devDependencies } = packageJson;
  const newDep = dependencies && dependencies[key];
  const newDevDep = devDependencies && devDependencies[key];

  return mergeAll([
    packageJson,
    newDep && {
      dependencies: {
        ...dependencies,
        [key]: publishedPackages[key],
      },
    },
    newDevDep && {
      devDependencies: {
        ...devDependencies,
        [key]: publishedPackages[key],
      },
    },
  ]);
};
```

and used in `updatePackageJson` like:

```js
  const newPackageJson = Object.keys(publishedPackages).reduce(
    mergeUpdates(publishedPackages),
    packageJson,
  );
```

more lines required to the same thing but possibly more readable and doesn't include mutations.

Reply:
I agree on the avoidance on mutation. But I think this sacrifices some readability to achieve that. I feel your code is more elegant but significantly less readable because of multiple complex uses of object destruction and the reduce function. By renaming some of the variables I think the updates I made should make it easier to read.